internal/jimm/model_cleanup.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2024 Canonical.
+
+package jimm
+
+import (
+	"context"
+	"fmt"
+
+	jujuparams "github.com/juju/juju/rpc/params"
+	"github.com/juju/juju/state"
+	"github.com/juju/zaputil/zapctx"
+	"go.uber.org/zap"
+
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/errors"
+)
+
+// CleanupModelsDying loops over dying models, contacting the respective controller.
+// And deleting the model from our database if the error is `NotFound` which means the model was successfully deleted.
+func (j *JIMM) CleanupModelsDying(ctx context.Context) error {
+	const op = errors.Op("jimm.WatchModelsDying")
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	err := j.DB().ForEachModel(ctx, func(m *dbmodel.Model) error {
```

Comment:
Instead of for each models, rather list models that are dying.

Reply:
I think foreach is fine honestly and dialing the controllers makes a lot of sense to me. Why would we have a db method?